source/funkin/ui/debug/charting/ChartEditorState.hx

```diff
@@ -3656,6 +3659,34 @@ class ChartEditorState extends UIState // UIState derives from MusicBeatState
         }
       }
 
+      // Retrieve notes stacked on top of others (TODO: Is there a non-O(n^2) way of doing this?)
```

Comment:
we could "cut" the notes into chunks an then do the check.
it would basically be O(n*m), (worst case O(n^2)

we could maybe also parallelize it

but i don't think it really matters, since usually there aren't that many notes.